ddtrace/contrib/pymongo/client.py

```diff
@@ -159,6 +169,18 @@ def send_message_with_response(self, operation, *args, **kwargs):
                 result = self.__wrapped__.send_message_with_response(operation, *args, **kwargs)
                 if result and result.address:
                     set_address_tags(span, result.address)
+                if result and self._is_query(operation):
+                    if VERSION < (3, 2, 0):
+                        docs = _unpack_response(response=result.data)
+                        span.set_metric(db.ROWCOUNT, docs["number_returned"])
+                    elif (3, 2, 0) <= VERSION < (3, 6, 0):
+                        docs = _unpack_response(response=result.data)
+                        cursor = docs["data"][0]["cursor"]
+                        set_query_rowcount(cursor=cursor, span=span)
+                    else:
+                        docs = result.data.unpack_response()
+                        cursor = docs[0]["cursor"]
```

Comment:
can we use docs[0]["number_returned"] here instead of the cursor? Also can docs be empty?

Reply:
I don't believe `number_returned` is defined for >3.6. Ill look into this

Reply:
to add to this, some of the tests already iterate through the whole cursor, i've added the same iteration to the added rowcount test so ensure the cursor is still alive and unaffected.